src/components/Layout/Header/Header.tsx

```diff
@@ -0,0 +1,150 @@
+import LangSwitch from '../../LangSwitch/LangSwitch';
+import { auth, logout } from '../../../firebase';
+import { useTranslation } from 'react-i18next';
+import React, { useEffect, useRef, useState } from 'react';
+import {
+  Group,
+  Text,
+  Button,
+  Flex,
+  Header,
+  useMantineTheme,
+  Burger,
+  Transition,
+  Paper,
+} from '@mantine/core';
+import { useDisclosure } from '@mantine/hooks';
+import { IconBrandGraphql } from '@tabler/icons-react';
+import { useAuthState } from 'react-firebase-hooks/auth';
+import { useHeaderStyles } from './HeaderStyles';
+import { ThemeSwitch } from '../../ThemeSwitch/ThemeSwitch';
+import { Link, useLocation, useNavigate } from 'react-router-dom';
+
+export const HeaderCustom = () => {
+  const [prevPos, setPrevPos] = useState(window.pageYOffset);
+  const [opened, { toggle, close }] = useDisclosure(false);
+  const theme = useMantineTheme();
+  const { t } = useTranslation();
+  const [user] = useAuthState(auth);
+  const headerRef = useRef<HTMLElement>(null);
+  const { classes } = useHeaderStyles();
+  const navigate = useNavigate();
+  const { pathname } = useLocation();
+
+  useEffect(() => {
+    const scrollHandler = () => {
+      const currentPos = window.pageYOffset;
+      const header = headerRef.current as HTMLElement;
+      header.style.height = prevPos > currentPos ? '80px' : '60px';
+      header.style.backgroundColor =
+        prevPos > currentPos
+          ? 'transparent'
+          : theme.colorScheme === 'dark'
+          ? theme.colors.dark[6]
+          : theme.colors.gray[5];
+      setPrevPos(currentPos);
+    };
+    window.addEventListener('scroll', scrollHandler);
+    return () => window.removeEventListener('scroll', scrollHandler);
+  }, [prevPos, theme.colorScheme, theme.colors.dark, theme.colors.gray]);
+
+  const handleNavigate = () => {
+    navigate('/auth');
+  };
```

Comment:
You can add close() to this function and reuse it in onClick. It will reduce duplications